From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ioana Pandele <ioanap@chromium.org>
Date: Thu, 9 Jan 2025 06:51:40 -0800
Subject: [PATCH] [UPMLoginDbSunset] Update buildflag checks for the exporter

Bug:378650395

Change-Id: Ia8e73baaf34552d5e8bac507438cbf033ab72492
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6163423
Reviewed-by: Anna Tsvirchkova <atsvirchkova@google.com>
Commit-Queue: Ioana Pandele <ioanap@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1404151}
---
 .../android/login_db_deprecation_runner_factory.cc             | 3 ++-
 .../browser/password_manager/profile_password_store_factory.cc | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/chrome/browser/password_manager/android/login_db_deprecation_runner_factory.cc b/chrome/browser/password_manager/android/login_db_deprecation_runner_factory.cc
index 4fe594963e88f..9c612b3309781 100644
--- a/chrome/browser/password_manager/android/login_db_deprecation_runner_factory.cc
+++ b/chrome/browser/password_manager/android/login_db_deprecation_runner_factory.cc
@@ -39,7 +39,7 @@ LoginDbDeprecationRunnerFactory::BuildServiceInstanceForBrowserContext(
     content::BrowserContext* context) const {
 #if BUILDFLAG(USE_LOGIN_DATABASE_AS_BACKEND)
   return nullptr;
-#endif
+#else
   Profile* profile = Profile::FromBrowserContext(context);
   PrefService* prefs = profile->GetPrefs();
 
@@ -62,4 +62,5 @@ LoginDbDeprecationRunnerFactory::BuildServiceInstanceForBrowserContext(
 
   return std::make_unique<password_manager::LoginDbDeprecationRunner>(
       profile->GetPrefs(), profile->GetPath());
+#endif  // BUILDFLAG(USE_LOGIN_DATABASE_AS_BACKEND)
 }
diff --git a/chrome/browser/password_manager/profile_password_store_factory.cc b/chrome/browser/password_manager/profile_password_store_factory.cc
index 03c6e51be2c1b..00b63fed191ae 100644
--- a/chrome/browser/password_manager/profile_password_store_factory.cc
+++ b/chrome/browser/password_manager/profile_password_store_factory.cc
@@ -101,7 +101,7 @@ scoped_refptr<RefcountedKeyedService> BuildPasswordStore(
 
   password_affiliation_adapter->RegisterPasswordStore(ps.get());
   affiliation_service->RegisterSource(std::move(password_affiliation_adapter));
-#if BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_ANDROID) && !BUILDFLAG(USE_LOGIN_DATABASE_AS_BACKEND)
   CHECK(password_manager_android_util::IsInternalBackendPresent());
   password_manager::LoginDbDeprecationRunner* login_db_deprecation_runner =
       LoginDbDeprecationRunnerFactory::GetForProfile(profile);
